app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -40,4 +41,8 @@ def validate_default_value
   def puppet?
     true
   end
+
+  def check_override_selected
+    errors.add(:override, _("must be true to edit the parameter")) unless self.override || (self.changes.size == 1 && self.changes["description"].present?) || self.new_record?
```

Comment:
to make this less ugly, I suggest adding this to ActiveRecord::Base (in active_record_extensions.rb)
```ruby
def only_changed?(field_name)
  self.changes.size == 1 && self.changes[field_name].present?
end
```